concepts/eventing/performance evaluation/nats-jetstream/results_prod_HA.md

```diff
@@ -0,0 +1,240 @@
+-> Total Events Received  by **Stream**      : 43,083 (i.e. 43,083 - 0) 
+-> Total Events Processed by **Consumer**    : 43,083 (i.e. 43,083 - 0)
+-> Total Events Received  by **Sink**        : 43,012 (There were some errors in sink for storing the data to database)
+```
+
+![](assets/ha_crash4_1.png "")
+![](assets/ha_crash4_2.png "")
+
+
+### Run ID: 13/6/2022T10:45 (Duration: 5m, Event Rate: 150rps)
+
+> **Note:** Eventing-controller pod was not deployed on the deleted node.
+
+**State before test run:**
+- Stream:
+  - LastSeq# 129,137
+  - Leader: eventing-nats-1
+- Consumer:
+  - Ack Floor: Stream sequence# 129,137
+
+**State after test run:**
+- Stream:
+  - LastSeq# 161,358
+  - Leader: eventing-nats-0
+- Consumer:
+  - Ack Floor: Stream sequence# 161,358
+  - Redelivered Messages: 0
+  - Unprocessed Messages: 0
+
+```
+-> Old Stream Leader                         : `eventing-nats-1`
+-> New Stream Leader                         : `eventing-nats-0`
+-> Total Events Sent      by **Test Sender** : 32,221 (+ 16 Failed = 32,237)
+-> Total Events Received  by **Stream**      : 32,221 (i.e. 161,358 - 129,137) 
+-> Total Events Processed by **Consumer**    : 32,221 (i.e. 161,358 - 129,137)
+-> Total Events Received  by **Sink**        : 32,132 (There were some errors in sink for storing the data to database)
+```
+
+![](assets/ha_crash5_1.png "")
+![](assets/ha_crash5_2.png "")
+![](assets/ha_crash5_3.png "")
+
+> **Note:** There were some subscription reconciliation errors logs in eventing-controller pod, but after sometime it was able to reconnect to NATS server and reconcile the subscription successfully.
+
+### Run ID: 9/6/2022T9:22 (Duration: 5m, Event Rate: 150rps)
+
+> **Note:** Eventing-controller pod was not deployed on the deleted node.
+
+**State after test run:**
+- Consumer:
+  - Redelivered Messages: 0
+  - Unprocessed Messages: 29,386
+
+```
+-> Old Stream Leader                         : `eventing-nats-2`
+-> New Stream Leader                         : `eventing-nats-1`
+-> Total Events Sent      by **Test Sender** : 41,956 (+ 1 Failed = 41,957)
+-> Total Events Received  by **Sink**        : 41,960
+```
+
+> **NOTE:** After the test run, the consumer info showed `Unprocessed Messages: 29,386` but the `Redelivered Messages` and `Outstanding Acks` were zero. This was because of a bug in NATS v2.8.2 and is fixed in new version v2.8.4 by [PR](https://github.com/nats-io/nats-server/pull/3148).
+
+![](assets/ha_crash6_5.png "")
+![](assets/ha_crash6_1.png "")
+![](assets/ha_crash6_2.png "")
+![](assets/ha_crash6_3.png "")
+![](assets/ha_crash6_4.png "")
+
+## Conclusion
+Based on the findings, its recommended to use atleast 3 stream replicas to provide high availibilty of stream in case of single-node failure. Therefore, now the production profile will use 3 replicas of stream and evaluation profile will still be set to use 1 stream replica ([related PR](https://github.com/kyma-project/kyma/pull/14539)).
```

Comment:
I think it would be helpful if you walk the reader through your thinking process and how you reached the conclusion :) It is also a good summary of the test results :)